lib/matplotlib/artist.py

```diff
@@ -903,6 +903,27 @@ def matchfunc(x):
             artists.append(self)
         return artists
 
+    def get_pixel_data(self, event):
+        """
+        Get the pixel data for a given event.
+        """
+        return None
+
+    def format_pixel_data(self, data):
+        """
+        Return *pixel data* string formatted.
+        """
+        try:
+            is_int = isinstance(data[0], int)
+        except (TypeError, IndexError):
+            is_int = isinstance(data, int)
+            data = [data]
+        if is_int:
+            data = ', '.join(item for item in data)
```

Comment:
How is this working? I don't think `is_int` is ever being set as this line should not work - `join`should give a TypeError if it gets a sequence of `int` and not a sequence of string. Try ```data=[1, 2, 3]; data = ', '.join(item for item in data)``` (BTW, you could have just used `data` instead of `item for item in data`).

I suspect what is going on in your example is that a numpy array is being given and `is_int` is being set to `False` and the else condition is always used. That being the case, then you might as well not even test for int and just use the `{:0.3g}` formatter in all cases.